From a5f1b41b0a10fc1dc64f17b677ddf14e15f8d40e Mon Sep 17 00:00:00 2001
From: "Brian C. Lane" <bcl@redhat.com>
Date: Tue, 23 Aug 2016 08:55:18 -0700
Subject: [PATCH 53/53] tests: Set optimal blocks to 64 for scsi_debug devices

The Linux kernel 4.5 changed the optimal blocks count from 64 to 1024
This causes tests using scsi_debug devices to fail because of alignment
issues. Set the opt_blks to 64 so that we have consistent behavior
across kernels.

(cherry picked from commit bd2664285e4014d2d4c80cee3e87609272e1fca9)
---
 tests/t-local.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tests/t-local.sh b/tests/t-local.sh
index a7d5226..13073d2 100644
--- a/tests/t-local.sh
+++ b/tests/t-local.sh
@@ -98,7 +98,7 @@ scsi_debug_setup_()
   # It is not trivial to determine the name of the device we're creating.
   # Record the names of all /sys/block/sd* devices *before* probing:
   touch stamp
-  modprobe scsi_debug "$@" || { rm -f stamp; return 1; }
+  modprobe scsi_debug opt_blks=64 "$@" || { rm -f stamp; return 1; }
   scsi_debug_modprobe_succeeded_=1
   test "$VERBOSE" = yes \
     && echo $ME_ modprobe scsi_debug succeeded 1>&2
-- 
2.7.4

